packages/modules/providers/auth-github/src/services/github.ts

```diff
@@ -192,16 +202,17 @@ export class GithubAuthService extends AbstractAuthModuleProvider {
     }
   }
 
-  private getRedirect({ clientId, callbackUrl }: LocalServiceConfig) {
+  private getRedirect(clientId: string, callbackUrl: string, stateKey: string) {
     const redirectUrlParam = `redirect_uri=${encodeURIComponent(callbackUrl)}`
     const clientIdParam = `client_id=${clientId}`
     const responseTypeParam = "response_type=code"
-
+    const stateKeyParam = `state=${stateKey}`
     const authUrl = new URL(
       `https://github.com/login/oauth/authorize?${[
         redirectUrlParam,
         clientIdParam,
         responseTypeParam,
+        stateKeyParam,
       ].join("&")}`
     )
```

Comment:
suggestion: What do you think of the following?

```suggestion
    const stateKeyParam = `state=${stateKey}`
    const authUrl = new URL('https://github.com/login/oauth/authorize')
    authUrl.searchParams.append('redirect_uri',  encodeURIComponent(callbackUrl)) 
    authUrl.searchParams.append('client_id',  clientId). 
    authUrl.searchParams.append('responseTypeParam',  'code')
    authUrl.searchParams.append('state',  stateKey)
```

if you agree and that works for you then we can cleanup those variables above as well

Reply:
Yup sounds good 👍